parquet-hadoop/src/main/java/org/apache/parquet/filter2/statisticslevel/StatisticsFilter.java

```diff
@@ -217,6 +219,70 @@ public <T extends Comparable<T>> Boolean visit(Contains<T> contains) {
     return contains.filter(this, (l, r) -> l || r, (l, r) -> l && r, v -> BLOCK_MIGHT_MATCH);
   }
 
+  @Override
+  public Boolean visit(Size size) {
+    final ColumnChunkMetaData metadata = getColumnChunk(size.getColumn().getColumnPath());
+    if (metadata == null) {
+      // the column isn't in this file, so fail eq/gt/gte targeting size > 0
+      final boolean blockCannotMatch =
+          size.filter((eq) -> eq > 0, (lt) -> false, (lte) -> false, (gt) -> gt >= 0, (gte) -> gte > 0);
+      return blockCannotMatch ? BLOCK_CANNOT_MATCH : BLOCK_MIGHT_MATCH;
+    }
+
+    final SizeStatistics stats = metadata.getSizeStatistics();
```

Comment:
Should we check whether the list type can be supported before proceeding? For example, perhaps we don't want to support legacy LIST-annotated 2-level structure and unannotated repeated field according to https://github.com/apache/parquet-format/pull/466.

Reply:
yes I think that makes sense. Maybe we can add support for those at a later date... but this PR is already complex enough as-is, I think. I'll add a check!

Reply:
We could also offer _only_ record-level filtering (`IncrementallyUpdatedFilterPredicate`) for 2-level and unannotated lists, since that doesn't depend on any particular list implementation. We'd just skip StatisticsFilter/BloomFilter/DictionaryFilter/ColumnIndexFilter. That way `size()` would still work for any kind of list--it would just be less less efficient.